src/Core/Foundation/PrestaShopVersion/PrestaShopVersion.php

```diff
@@ -0,0 +1,214 @@
+
+    /**
+     * Returns if the current PrestaShop version is greater than or equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isGreaterThanOrEqualTo($version)
+    {
+        return $this->versionCompare($version, '>=');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is less than the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isLessThan($version)
+    {
+        return $this->versionCompare($version, '<');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is less than or equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isLessThanOrEqualTo($version)
+    {
+        return $this->versionCompare($version, '<=');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isEqualTo($version)
+    {
+        return $this->versionCompare($version, '=');
+    }
+
+    /**
+     * Returns if the current PrestaShop version is not equal to the provided version
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    public function isNotEqualTo($version)
+    {
+        return $this->versionCompare($version, '!=');
+    }
+
+    /**
+     * Compares the current PrestaShop version with the provided version depending on the provided operator
+     *
+     * @param $version Must be a valid PrestaShop version string, for example "1.7.4.0"
+     * @param $operator Operator for version_compare(), allowed values are: <, lt, <=, le, >, gt, >=, ge, ==, =, eq, !=, <>, ne
+     *
+     * @return boolean Result of the comparison
+     *
+     * @throws InvalidArgumentException If the provided version is invalid
+     */
+    private function versionCompare($version, $operator)
+    {
+        try {
+            $this->checkVersion($version);
+        } catch(InvalidVersionException $e) {
+            throw new InvalidArgumentException($e->getMessage());
+        }
+
+        return version_compare(AppKernel::VERSION, $version, $operator);
```

Comment:
set `AppKernel::VERSION` as parameter in constructor, so everyone will be able to use this =)

Reply:
@PierreRambaud The method is private and not meant to be used outside the class. There are quite some other methods to compare a version string to the actual PS version. Is this ok?